### ISSUE TITLE
Feature Request: Provide template helpers for formatting dates

### ISSUE THREAD

**Issue by pichfl** [willrax/ember-luxon#3]
ember-moment provides a range of helpers:
https://github.com/stefanpenner/ember-moment#helpers

```hbs
{{moment '12-25-1995' 'MM-DD-YYYY'}}
{{moment-format '12-25-1995' 'MM/DD/YYYY' 'MM-DD-YYYY'}} {{!-- outputFormat and inputFormat are optional --}}
{{moment-from-now (now) hideSuffix=true}} {{!-- hideSuffix is optional --}}
{{moment-to-now (unix timeStamp) date hidePrefix=true}} {{!-- hidePrefix is optional --}}
{{moment-duration number units}} {{!-- units is optional --}}
{{moment-calendar date referenceDate}} {{!-- reference date is optional --}}
{{moment-diff dateA dateB precision='day' float=true}} {{!-- precision is optional, float is optional --}}
{{is-before date comparison precision='year'}} {{!-- precision is optional --}}
{{is-after date comparison precision='year'}} {{!-- precision is optional --}}
{{is-same date comparison precision='year'}} {{!-- precision is optional --}}
{{is-same-or-before date comparison precision='year'}} {{!-- precision is optional --}}
{{is-same-or-after date comparison precision='year'}} {{!-- precision is optional --}}
{{is-between date comparisonA comparisonB precision='year' inclusivity='[)'}} {{!-- precision is optional, inclusivity optional  --}}
```

ember-luxon should provide similar drop-in replacements to make it easier to switch.

- [x] `{{luxon '12-25-1995' 'MM-DD-YYYY'}}`
- [x] `{{luxon-format '12-25-1995' 'MM/DD/YYYY' 'MM-DD-YYYY'}} {{!-- outputFormat and inputFormat are optional --}}`
- [ ] `{{luxon-duration number units}} {{!-- units is optional --}}`
- [x] `{{luxon-diff dateA dateB precision='day' float=true}} {{!-- precision is optional, float is optional --}}`
- [x] `{{is-before date comparison precision='year'}} {{!-- precision is optional --}}`
- [x] `{{is-after date comparison precision='year'}} {{!-- precision is optional --}}`
- [x] `{{is-same date comparison precision='year'}} {{!-- precision is optional --}}`
- [ ] `{{is-same-or-before date comparison precision='year'}} {{!-- precision is optional --}}`
- [ ] `{{is-same-or-after date comparison precision='year'}} {{!-- precision is optional --}}`
- [x] `{{is-between date comparisonA comparisonB precision='year' inclusivity='[)'}} {{!-- precision is optional, inclusivity optional  --}}`

Not supported by Luxon:
```hbs
{{moment-calendar date referenceDate}} {{!-- reference date is optional --}}
{{moment-from-now (now) hideSuffix=true}} {{!-- hideSuffix is optional --}}
{{moment-to-now (unix timeStamp) date hidePrefix=true}} {{!-- hidePrefix is optional --}}
```

**Comment by willrax** [comment]
Hey @pichfl! We use these a lot on our application, so it's definitely on the list. I'll probably get some time to implement these over the next week.

**Comment by pichfl** [comment]
Turns out this will be a little more difficult than expected as Luxon does not support humanization yet. We'll have to look into other options.

https://moment.github.io/luxon/docs/manual/moment.html#humanization

**Comment by pichfl** [comment]
Maaaaaaybe it might be wise to leave the humanization to other libraries such as https://date-fns.org or https://github.com/ember-intl/ember-intl#format-time

**Comment by willrax** [comment]
I agree. I think it's fine for people to bring in other libraries to perform things that luxon doesn't include.

**Comment by willrax** [comment]
I think these ones:

```hbs
{{moment '12-25-1995' 'MM-DD-YYYY'}}
{{moment-format '12-25-1995' 'MM/DD/YYYY' 'MM-DD-YYYY'}}
{{moment-diff dateA dateB precision='day'}}
{{is-before date comparison precision='year'}} {{!-- precision is optional --}}
{{is-after date comparison precision='year'}} {{!-- precision is optional --}}
{{is-same date comparison precision='year'}}
{{is-same-or-before date comparison precision='year'}}
{{is-same-or-after date comparison precision='year'}}
{{is-between date comparisonA comparisonB precision='year'}} 
```

Should be ok

**Comment by pichfl** [comment]
I added todos for the various helpers to the description of this issue.

**Comment by willrax** [comment]
I'm back from holidays now 🎉  and will be adding these over the weekend.

**Comment by willrax** [comment]
released in v1.3.0
